components/AutoComplete.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from "react";
+import "tailwindcss/tailwind.css";
+
+interface AutoCompleteProps {
+  value?: string;
+  suggestions?: string[];
+  onChange?: (value: string) => void | undefined;
+  onSelect?: (value: string) => void | undefined;
+}
+
+const AutoComplete = ({
+  value = "",
+  suggestions = [],
+  onChange = () => {},
+  onSelect = () => {},
+}: AutoCompleteProps) => {
+  const [isFocused, setIsFocused] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+
+  const suggestionsSize = suggestions.length;
+  const currentValue = suggestions[selectedIndex] || value;
+
+  useEffect(() => {
+    setSelectedIndex(-1);
+  }, [value, suggestionsSize]);
+
+  const onKeyDown = (e: React.KeyboardEvent) => {
```

Comment:
Passing the `HTMLInputElement` as a generic then we don't have to us casting in the subsequent code.
```suggestion
  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
```